surrealdb/http.py

```diff
@@ -203,13 +203,13 @@ async def select(self, thing: str) -> List[Dict[str, Any]]:
             Select a specific record from a table (or other entity)
                 person = await db.select('person:h5wxrf2ewk8xjxosxtyc')
         """
-        if ":" in thing:
-            table, id = thing.split(":")
-            response = await self._request(method="GET", uri=f"/key/{table}/{id}")
-            if not response:
-                raise SurrealException(f"Key {id} not found in table {table}")
-        else:
-            response = await self._request(method="GET", uri=f"/key/{thing}")
+        table, record_id = thing.split(":") if ":" in thing else (thing, None)
+        response = await self._request(
+            method="GET",
+            uri=f"/key/{table}/{record_id}" if record_id else f"/key/{table}",
+        )
+        if not response and record_id:
```

Comment:
```suggestion
        if not response and record_id is not None:
```

Reply:
Thank you for your review, @CaedenPH. I agree with your suggestion of taking inspiration from the Mongodb Python library. We could use the following code as an example:

```python
surrealdb = SurrealHTTP(**config)
table = surrealdb["person"]
table.create({
    'name': 'Tobie',
    'settings': {
        'active': True,
        'marketing': True,
    },
})

table.update({
    'name': 'Tobie',
    'settings': {
        'active': True,
        'marketing': True,
    },
})
```

I think it makes more sense to perform different types of operations this way. Let me know if you have any further thoughts or suggestions.